spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/adapter/SpringDynamoDBAdapterClient.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.kinesis.adapter;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBStreams;
+import com.amazonaws.services.dynamodbv2.model.AmazonDynamoDBException;
+import com.amazonaws.services.dynamodbv2.streamsadapter.AmazonDynamoDBStreamsAdapterClient;
+import com.amazonaws.services.kinesis.model.DescribeStreamRequest;
+import com.amazonaws.services.kinesis.model.InvalidArgumentException;
+import com.amazonaws.services.kinesis.model.ListShardsRequest;
+import com.amazonaws.services.kinesis.model.ListShardsResult;
+import com.amazonaws.services.kinesis.model.ResourceNotFoundException;
+import com.amazonaws.services.kinesis.model.StreamDescription;
+
+/**
+ * This is Spring Cloud DynamoDB Adapter to be able to support {@code ListShards} operations.
+ *
+ * @see <a href="https://docs.aws.amazon.com/kinesis/latest/APIReference/API_ListShards.html">ListShards</a>
+ *
+ * @author acm
+ */
+public class SpringDynamoDBAdapterClient extends AmazonDynamoDBStreamsAdapterClient {
+	private static final String SEPARATOR = "!!##%%";
+
+	public SpringDynamoDBAdapterClient(AmazonDynamoDBStreams amazonDynamoDBStreams) {
+		super(amazonDynamoDBStreams);
+	}
+
+	/**
+	 * List shards for a DynamoDB Stream using its <@code DescribeStream> API, as they don't support
+	 * <@code ListShards> operations. Returns the result adapted to use the AmazonKinesis model.
+	 *
```

Comment:
No blank lines in the method level JavaDocs.

Reply:
Done, but is this a Spring projects convention? [Javadoc recommends](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html): `Insert a blank comment line between the description and the list of tags, as shown.`

Reply:
That is Oracle recommendations.
Ours are here: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting

Reply:
:+1: